libimage/manifest_list.go

```diff
@@ -325,8 +325,19 @@ func (m *ManifestList) Add(ctx context.Context, name string, options *ManifestLi
 	if err != nil {
 		withDocker := fmt.Sprintf("%s://%s", docker.Transport.Name(), name)
 		ref, err = alltransports.ParseImageName(withDocker)
+		if err == nil {
+			var src types.ImageSource
+			src, err = ref.NewImageSource(ctx, nil)
+			if err == nil {
+				src.Close()
+			}
+		}
 		if err != nil {
```

Comment:
did you mean for this conditional to handle the alltransports.parseimagename from earlier ?

Reply:
Yes, I expect alltransports.ParseImageName() to fail when given a string that consists of "docker://" followed by exactly 64 hex digits.